web/src/elements/wizard/TypeCreateWizardPage.ts

```diff
@@ -0,0 +1,136 @@
+import "@goauthentik/admin/common/ak-license-notice";
+import { WithLicenseSummary } from "@goauthentik/elements/Interface/licenseSummaryProvider";
+import { WizardPage } from "@goauthentik/elements/wizard/WizardPage";
+
+import { msg, str } from "@lit/localize";
+import { CSSResult, TemplateResult, css, html, nothing } from "lit";
+import { property, state } from "lit/decorators.js";
+
+import PFCard from "@patternfly/patternfly/components/Card/card.css";
+import PFForm from "@patternfly/patternfly/components/Form/form.css";
+import PFRadio from "@patternfly/patternfly/components/Radio/radio.css";
+import PFGrid from "@patternfly/patternfly/layouts/Grid/grid.css";
+import PFBase from "@patternfly/patternfly/patternfly-base.css";
+
+import { TypeCreate } from "@goauthentik/api";
+
+export enum TypeCreateWizardPageLayouts {
+    list,
+    grid,
+}
+
+export abstract class TypeCreateWizardPage extends WithLicenseSummary(WizardPage) {
```

Comment:
The only thing I could wish is that... this only shows a collection of radios.  I could wish is was relatively standalone and *evented* rather than relying on a pair of callbacks for its behavior.  Then it wouldn't need all that inheritance; you could just `<ak-type-create-wizard-page .types=${this.providerTypes} ?hidden={currentPage == 0} @select=${ ... set host type }></ak-type-create-wizard-page>`.

Reply:
The initial implementation worked like that, however it was quite a bit more annoying to use as with how the page is used there isn't a direct reference to the wizard host which would be required to advance pages...although I suppose that could be fixed with an `@query` attribute

Reply:
Yeah, the current architecture is much more OO oriented, less like traditional HTML.  It's fine, it works, and I have long learned to say "That's not how I would do it, but it gets the job done and it's not wrong."